request-manager/graph/resolver.go

```diff
@@ -240,74 +212,72 @@ func (o *resolver) buildSequence(cfg buildSequenceConfig) (*Graph, error) {
 			}
 
 			// Add the "if" to the node args if it's present
```

Comment:
This only needs to be done if `n.IsConditional()` (next block), right?

Reply:
`each:` is for sequence expansion, not conditional nodes. Conditional nodes are `if:` and `eq:`.

Reply:
Yes, but his isn't `each:`. What I'm suggesting is:
```go
			if nodeSpec.IsConditional() {
				// Node is a conditional: choose which path to
				// take, and recursively build the subgraph

				// Add the "if" to the node args if it's present
				if nodeSpec.If != nil {
					if ifArg, ok := jobArgs[*nodeSpec.If]; ok {
						jobArgsCopy[*nodeSpec.If] = ifArg
					}
				}
```
because `nodeSpec.If` is only valid if the node is a conditional.